libs/api/libbpf_link.cpp

```diff
@@ -6,8 +6,8 @@
 #include "libbpf.h"
 #include "libbpf_internal.h"
 
-int
-bpf_link__pin(struct bpf_link* link, const char* path)
+static int
+_bpf_link__pin(struct bpf_link* link, const char* path) noexcept
```

Comment:
This function calls `ebpf_object_pin()` which already is marked as `noexcept`. So do we need to have this function defined?

Reply:
Yes that would be a good direction.   The upstream libbpf is only in C so the more we stick to C the easier it will be to start using the upstream as in issue #351.

Reply:
Agree. I think a lot of the libbpf APIs are in already in C, so for those APIs we can skip creating a copy, and only create a copy for the APIs are in C++ (or is not possible to change to C)

Reply:
OK, that makes sense. I will abandon this PR, rework this and create a new PR (probably after thanksgiving at this point).